server/models.py

```diff
@@ -404,7 +405,24 @@ class Batch(BaseModel):
         order_by="BatchResultTallySheet.created_at",
     )
 
-    __table_args__ = (UniqueConstraint("jurisdiction_id", "tabulator", "name"),)
+    # Only one of the following should be populated, enforced via a check constraint defined below
+    last_edited_by_support_user_email = Column(String(200))
+    last_edited_by_jurisdiction_admin_email = Column(String(200))
+    last_edited_by_tally_entry_user_id = Column(
+        String(200), ForeignKey("tally_entry_user.id", ondelete="set null")
+    )
+
+    last_edited_by_tally_entry_user = relationship("TallyEntryUser")
+
+    __table_args__ = (
+        UniqueConstraint("jurisdiction_id", "tabulator", "name"),
+        CheckConstraint(
+            "(cast(last_edited_by_support_user_email is not null as int) +"
+            " cast(last_edited_by_jurisdiction_admin_email is not null as int) +"
+            " cast(last_edited_by_tally_entry_user_id is not null as int)) <= 1",
+            "only_one_of_last_edited_by_fields_is_specified_check",
+        ),
```

Comment:
Addressed one of my main concerns about this alternative schema: That we have to somehow make sure that only one of the three `last_edited_by` fields is specified.

This constraint will now guarantee that! Manually verified that it works

Reply:
Wow cool!